lldb/tools/lldb-dap/ProtocolUtils.cpp

```diff
@@ -105,6 +105,21 @@ protocol::Source CreateSource(lldb::SBAddress address, lldb::SBTarget &target) {
   return CreateSource(line_entry.GetFileSpec());
 }
 
+protocol::Source CreateSource(lldb::SBFrame frame) {
+  if (!frame.IsValid())
+    return {};
```

Comment:
I think we should make this function return an `std::optional<protocol::Source>` as I think this should only be set if the source is valid. The source should have at least `path` and/or `sourceReference` set.